tests/api/api-resources.test.js

```diff
@@ -0,0 +1,181 @@
+// Copyright Contributors to the Open Cluster Management project
+
+jest.retryTimes(global.retry)
+
+const { fail } = require('assert')
+const { performance } = require('perf_hooks')
+
+const squad = require('../../config').get('squadName')
+const {
+  getKubeConfig,
+  getSearchApiRoute,
+  getToken,
+  searchQueryBuilder,
+  sendRequest,
+  clusterLogin,
+} = require('../common-lib/clusterAccess')
+
+const {
+  fetchAPIResourcesWithListWatchMethods,
+  formatResourcesFromSearch,
+  formatFilters,
+  getMismatchedResources,
+  getResourcesFromOC,
+  getClusterList,
+  shouldUseAPIGroup,
+  verifyMissingResourcesFound,
+} = require('../common-lib/index')
+
+const { sleep } = require('../common-lib/sleep')
+
+// Set list to ignore resources that aren't being collected by Search.
+// When using the oc command clusterclaim doesn't include the namespace, therefore, for testing purposes, we will omit that resource object.
+const ignoreKindResourceList = ['clusterclaim', 'event']
+
+// Set list of resources that require filtering by api group.
+const requireAPIGroup = []
+
+/**
+ * Base test for kubernetes
+ * @param {*} kind The resource object kind that will be used for testing.
+ * @param {*} apigroup The apigroup of the object kind.
+ * @param {*} cluster The cluster of the object kind.
+ * @param {*} namespace The namespace of the object kind.
+ */
+function baseTest(
+  kind,
+  apigroup,
+  cluster = { type: 'hub', name: 'local-cluster' },
+  namespace = '--all-namespaces'
+) {
+  var runTest = test
+  var expectedResources = []
+
+  try {
+    expectedResources = getResourcesFromOC(kind, apigroup, namespace, cluster)
+  } catch (err) {
+    console.error(
+      `Skipping test because there was an error getting the expected resources from OC command. kind=${kind} apigroup=${apigroup.name} namespace=${namespace} cluster=${cluster.name} Error: ${err}`
+    )
+    runTest = test.skip
+  }
+
+  runTest(
+    `[P2][Sev2][${squad}] verify data integrity for resource property: ${
+      !apigroup.useAPIGroup ? kind : `${kind}.${apigroup.name}`
+    }`,
+    async () => {
+      const filters = formatFilters(kind, apigroup, namespace, cluster)
+      var mismatchResources = []
+
+      // Fetch data from the search api.
+      var query = searchQueryBuilder({ filters })
+
+      // Monitor how long search took to return results.
+      var startTime = performance.now()
+      var resp = await sendRequest(query, token)
+      var endTime = performance.now()
+      var totalElapsedTime = endTime - startTime
+
+      if (totalElapsedTime > 30000) {
```

Comment:
All the times in this file should be configurable with env variables.
It's very likely that we'll need to tweak these to optimize the test.

(You can make this change in the next PR.)